packages/mgt-components/src/components/mgt-file-list/mgt-file-list.ts

```diff
@@ -648,11 +650,23 @@ export class MgtFileList extends MgtTemplatedComponent {
       );
     }
 
-    if (this.pageIterator.hasNext) {
-      await this.pageIterator.next();
-      this.files = this.pageIterator.value;
-      this.requestUpdate();
+    // render next page from cache if exists, or else use iterator
+    if (this._files.length > 0) {
+      if (this._files.length >= this.pageSize) {
+        this.files = this.files.concat(this._files.slice(0, this.pageSize));
+        this._files = this._files.slice(this.pageSize);
+      } else {
+        this.files = this.files.concat(this._files.slice(0, this._files.length));
+        this._files = [];
+      }
```

Comment:
You could simplify this:

```suggestion
      this.files = [...this.files, ...this._files.splice(0, Math.min(this.pageSize, this._files.lenght))]
```